runtime/gc_glue_java/ObjectModel.hpp

```diff
@@ -180,7 +181,11 @@ class GC_ObjectModel : public GC_ObjectModelBase
 		{
 			UDATA classFlags = J9CLASS_FLAGS(clazz) & (J9AccClassReferenceMask | J9AccClassGCSpecial | J9AccClassOwnableSynchronizer);
 			if (0 == classFlags) {
-				result = SCAN_MIXED_OBJECT;
+				if (0 != clazz->selfReferencingField1) {
```

Comment:
I'm fine with this for this PR, but we should try an approach with a separate class flag.